### PR TITLE
azure_rm_webapp: fix updating app settings

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

fix  webapp module bug: updating app_settings is not done by create_or_update, but a specific  api.

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by yungezz** [comment]
@zikalino, could you pls help to review?

File: lib/ansible/modules/cloud/azure/azure_rm_webapp.py [hunk 1]
```diff
@@ -745,6 +746,10 @@ def exec_module(self, **kwargs):
 
                 self.results['id'] = response['id']
 
+            if self.to_do == Actions:UpdateAppSettingss:
```
Comment (gundalow) [inline lib/ansible/modules/cloud/azure/azure_rm_webapp.py]: I think there is a syntax error here.
```suggestion
            if self.to_do == Actions.UpdateAppSettings:
```

Has this been tested?

File: lib/ansible/modules/cloud/azure/azure_rm_webapp.py [hunk 2]
```diff
@@ -708,11 +708,12 @@ def exec_module(self, **kwargs):
                 if self.purge_app_settings:
                     to_be_updated = True
                     self.app_settings_strDic.properties = dict()
+                    self.to_do = Actions.UpdateAppSettingss
```
Comment (gundalow) [inline lib/ansible/modules/cloud/azure/azure_rm_webapp.py]:
```suggestion
                    self.to_do = Actions.UpdateAppSettings
```

**Comment by Fred-sun** [comment]
@yungezz need you update

File: lib/ansible/modules/cloud/azure/azure_rm_webapp.py (hunk 1 above)
Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_webapp.py]: sorry for the quick pr. I've updated after testing.

**Comment by Fred-sun** [comment]
@yuwzho @zikalino  need your review

File: lib/ansible/modules/cloud/azure/azure_rm_webapp.py [hunk 3]
```diff
@@ -708,11 +708,12 @@ def exec_module(self, **kwargs):
                 if self.purge_app_settings:
                     to_be_updated = True
                     self.app_settings_strDic.properties = dict()
+                    self.to_do = Actions.UpdateAppSettings
 
                 # check if app settings changed
                 if self.purge_app_settings or self.is_app_settings_changed():
                     to_be_updated = True
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_webapp.py:715]: so is it possible to update webapp and app settings at once?
seems like UpdateAppSettings overrides CreateOrUpdate, so if user wants to update both, only  settings will be updated

**Comment by Fred-sun** [comment]
@yungezz @zikalino How about the PR going ahead? need further review or merged,  I have tested the new module. It can be create, delete and purge app_settings.  Thanks!

**Comment by yungezz** [comment]
ready_for_review

File: lib/ansible/modules/cloud/azure/azure_rm_webapp.py [hunk 4]
```diff
@@ -504,7 +504,7 @@ def __init__(self):
             id=None,
         )
         self.state = None
-        self.to_do = Actions.NoAction
+        self.to_do = []
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_webapp.py:507]: So now "NoAction" and "Delete" constants defined above are not used at all. So I think they should be deleted.
Or maybe just use Boolean variables?

**Review by zikalino — APPROVED** [review]

**Pull request merged**